### ISSUE TITLE
Error with path="optimal"

### ISSUE THREAD

**Issue by rmcgibbo** [dgasmith/opt_einsum#2]
When running the following example, I get an exception
```
contract('ia,aj,ka->ijk', A, B, C, path="optimal")
```

The traceback is 
```
Traceback (most recent call last):
  File "opt_einsum.py", line 477, in <module>
    x, y = contract('ia,aj,ka->ijk', A, B, C, path="optimal")
  File "opt_einsum.py", line 425, in contract
    tmp_inputs.append(input_list.pop(x))
IndexError: pop index out of range
```

**Comment by dgasmith** [comment]
Can you give the shape of the arrays and branch?

Opt_einsum really should not be trying to build intermediates for that contraction unless one of the `ijk` indices is the sum of the other two.

**Comment by rmcgibbo** [comment]
All three were 5 x 5

**Comment by dgasmith** [comment]
Fixed in f5bcd34. Also brings up another operation type to consider optimizing.
